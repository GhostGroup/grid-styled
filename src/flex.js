import React from "react";
import styled from "styled-components";
import { responsiveStyle } from "styled-system";
import { oneOfType, number, string, array, bool } from "prop-types";
import propTypes from "./prop-types";
import Box from "./box";

const flexWrap = responsiveStyle("flex-wrap", "flexWrap", "wrap");
const flexDirection = responsiveStyle("flex-direction", "flexDirection");
const alignItems = props => responsiveStyle("align-items", "alignItems");
const justifyContent = props =>
  responsiveStyle("justify-content", "justifyContent");
const column = props => (props.column ? `flex-direction:column;` : null);

const Flex = styled(Box)(
  [],
  { display: "flex" },
  flexWrap,
  column,
  flexDirection,
  alignItems,
  justifyContent
);
Flex.displayName = "Flex";

const responsivePropType = oneOfType([number, string, array, bool]);

Flex.propTypes = Object.assign({}, propTypes, {
  flexWrap: responsivePropType,
  flexDirection: responsivePropType,
  alignItems: responsivePropType,
  justifyContent: responsivePropType,
  column: bool
});

export default Flex;
