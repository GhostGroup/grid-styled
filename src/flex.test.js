import React from "react";
import { shallow, mount } from "enzyme";
import Flex from "./flex";

describe("flex", () => {
  it("renders", () => {
    const tree = shallow(<Flex />);
    expect(tree).toMatchSnapshot();
  });

  it("renders with props", () => {
    const tree = shallow(
      <Flex
        flexWrap
        flexDirection="column"
        alignItems="center"
        justifyContent="space-between"
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it("renders with column prop", () => {
    const tree = shallow(<Flex column />);
    expect(tree).toMatchSnapshot();
  });

  it("renders with responsive props", () => {
    const tree = shallow(
      <Flex
        flexWrap={[true, false]}
        flexDirection={["column", "row"]}
        alignItems={["stretch", "center"]}
        justifyContent={["space-between", "center"]}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('can use a different component with the "is" prop', () => {
    const tree = mount(
      <Flex m={2} px={3} is="section">
        Test
      </Flex>
    );
    expect(tree).toMatchSnapshot();
  });
});
