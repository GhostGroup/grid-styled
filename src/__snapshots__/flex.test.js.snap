// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flex can use a different component with the "is" prop 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  box-sizing: border-box;
  margin: 16px;
  padding-left: 32px;
  padding-right: 32px;
}

<Flex
  is="section"
  m={2}
  px={3}
>
  <Box
    className="c0"
    is="section"
    m={2}
    px={3}
  >
    <styled.section
      className="c0"
      m={2}
      px={3}
    >
      <section
        className="c0 c1"
      >
        Test
      </section>
    </styled.section>
  </Box>
</Flex>
`;

exports[`flex renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<Box
  className="c0"
  is="div"
/>
`;

exports[`flex renders with column prop 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<Box
  className="c0"
  column={true}
  is="div"
/>
`;

exports[`flex renders with props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<Box
  alignItems="center"
  className="c0"
  flexDirection="column"
  flexWrap={true}
  is="div"
  justifyContent="space-between"
/>
`;

exports[`flex renders with responsive props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media screen and (min-width:40em) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<Box
  alignItems={
    Array [
      "stretch",
      "center",
    ]
  }
  className="c0"
  flexDirection={
    Array [
      "column",
      "row",
    ]
  }
  flexWrap={
    Array [
      true,
      false,
    ]
  }
  is="div"
  justifyContent={
    Array [
      "space-between",
      "center",
    ]
  }
/>
`;
